routes/runtime/github.rb

```diff
@@ -113,7 +118,12 @@ class CloverRuntime
         fail CloverError.new(400, "InvalidRequest", "Wrong parameters") if etags.nil? || etags.empty? || upload_id.nil? || size == 0
 
         entry = GithubCacheEntry[repository_id: repository.id, upload_id: upload_id, committed_at: nil]
-        fail CloverError.new(204, "NotFound", "No cache entry") if entry.nil? || entry.size != size
+        fail CloverError.new(204, "NotFound", "No cache entry") if entry.nil?
+
+        # If the size can not be set with reserveCache, we set it here.
+        if entry.size.nil?
+          entry.update(size: size)
+        end
```

Comment:
nit: 
```suggestion
        entry.update(size: size) if entry.size.nil?
```

Reply:
btw we already send an update query to database in the next lines: `entry.update(committed_at: Time.now)`. Can you combine them?

Reply:
I prefer to keep it explicit after checking whether the size is nil.

Reply:
I agree with to keep it explicit. You can combine them like 
```ruby
updates = {committed_at: Time.now}
updates[:size] = size if entry.size.nil?
entry.update(updates)
```